### DISCUSSION TITLE
"TypeError" while launching the webui

### DISCUSSION THREAD

**Discussion by loliana232** [vladmandic/sdnext#2737] [Q&A]
Hello, I'm trying to run sdxl with  SD.next webui on Intel gpu with openvino support, I followed all the instructions but it gives me this error every time while launching, I was using my pc and now I'm using "intel developer cloud instances" which is way better, but I got the same error with the both.

 *THE_ERROR*  :  

/home/u6dce01f3/Training/AI/GenAI/automatic 05:50:23-947021 INFO Starting SD.Next 05:50:23-951896 INFO Logger: file="/home/u6dce01f3dc9fcb2c1094b576fc96a24/Training/A I/GenAI/automatic/sdnext.log" level=INFO size=45049 mode=append 05:50:23-952827 INFO Python 3.9.16 on Linux 05:50:23-992835 INFO Version: app=sd.next updated=2024-01-13 hash=5fb290f url=https://github.com/vladmandic/automatic/tree/master 05:50:24-191590 INFO Platform: arch=x86_64 cpu=x86_64 system=Linux release=5.15.0-91-generic python=3.9.16 05:50:24-196849 INFO Skipping Torch tests 05:50:24-691986 INFO Extensions: disabled=[] 05:50:24-694309 INFO Extensions: enabled=['sd-webui-agent-scheduler', 'sd-extension-chainner', 'stable-diffusion-webui-rembg', 'sd-webui-controlnet', 'stable-diffusion-webui-images-browser', 'sd-extension-system-info', 'Lora'] extensions-builtin 05:50:24-721363 INFO Extensions: enabled=[] extensions 05:50:24-724338 INFO Startup: quick launch 05:50:24-726011 INFO Verifying requirements 05:50:24-732342 INFO Verifying packages 05:50:24-736849 INFO Extensions: disabled=[] 05:50:24-738661 INFO Extensions: enabled=['sd-webui-agent-scheduler', 'sd-extension-chainner', 'stable-diffusion-webui-rembg', 'sd-webui-controlnet', 'stable-diffusion-webui-images-browser', 'sd-extension-system-info', 'Lora'] extensions-builtin 05:50:24-745095 INFO Extensions: enabled=[] extensions 05:50:24-758538 INFO Extension preload: {'extensions-builtin': 0.01, 'extensions': 0.0} 05:50:24-762883 INFO Command line args: ['--skip-torch', '--use-openvino', '--listen', '--theme', 'dark', '--share'] share=True listen=True
theme=dark use_openvino=True 
skip_torch=True 
05:50:32-976348 INFO Load packages: torch=2.1.1+cu121 
diffusers=0.25.0 
gradio=3.43.2 
INFO:nncf:NNCF initialized successfully. Supported frameworks 
detected: torch, tensorflow, onnx, openvino
05:50:35-291935 INFO Engine: backend=Backend.DIFFUSERS 
compute=openvino 
mode=no_grad device=cpu 
cross-optimization="Scaled-Dot-Product" 
05:50:35-484813 INFO Device: device="Intel(R) Data Center GPU Max 
1100 (dGPU)"
openvino=2023.2.0 
2024-01-19 05:50:36.312809: E 
tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:9342] 
Unable to register cuDNN factory: Attempting to register factory for plugin 
cuDNN when one has already been registered
2024-01-19 05:50:36.312840: E 
tensorflow/compiler/xla/stream_executor/cuda/cuda_fft.cc:609] Unable 
to register cuFFT factory: Attempting to register factory for plugin 
cuFFT when one has already been registered
2024-01-19 05:50:36.312867: E 
tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:1518] Unable
to register cuBLAS factory: Attempting to register factory for plugin 
cuBLAS when one has already been registered
2024-01-19 05:50:39.573680: W 
itex/core/wrapper/itex_gpu_wrapper.cc:32] Could not load dynamic 
library: libsycl.so.7: cannot open shared object file: No such file or
directory
2024-01-19 05:50:39.624026: I 
itex/core/wrapper/itex_cpu_wrapper.cc:60] Intel Extension for 
Tensorflow* AVX512 CPU backend is loaded.
2024-01-19 05:50:39.664308: I itex/core/graph/xpu_graph.cc:130] Please
enable TF_ENABLE_ONEDNN_OPTS.So can benefit from the optimization of 
the MklCPUAllocator memory allocation.
2024-01-19 05:50:39.664367: E 
itex/core/wrapper/itex_gpu_wrapper.cc:49] Could not load Intel 
Extension for Tensorflow* GPU backend, GPU will not be used.
If you need help, create an issue at https://github.com/intel/intel-
extension-for-tensorflow/issues
2024-01-19 05:50:39.664587: E 
itex/core/wrapper/itex_gpu_wrapper.cc:49] Could not load Intel 
Extension for Tensorflow* GPU backend, GPU will not be used.
If you need help, create an issue at https://github.com/intel/intel-
extension-for-tensorflow/issues
05:50:41-123611 INFO Available VAEs: path="models/VAE" items=0 
05:50:41-126840 INFO Disabled extensions: ['sd-webui-controlnet'] 
05:50:41-501504 INFO Available models: path="models/Stable-
diffusion" 
items=2 time=0.37 
05:50:41-846367 INFO Extension: script='scripts/faceid.py' 
[2;36m05:50:41-842826[0m[2;36m [0m[1;31mERROR
[0m 
FaceID: No module named [32m'insightface'[0m 
05:50:42-749296 INFO Extension: 
script='extensions-builtin/sd-webui-agent-
scheduler/scr
ipts/task_scheduler.py' Using sqlite file: extensions-builtin/sd-webui-agent-scheduler/task_schedu
ler.sqlite3 
05:50:42-962746 INFO Extensions time: 1.25 { automatic=0.16 
Lora=0.30 
sd-extension-chainner=0.10 
sd-webui-agent-scheduler=0.46 
stable-diffusion-webui-images-browser=0.21 } 
05:50:43-042495 INFO Load UI theme: name="black-teal" style=Auto 
base=sdnext.css 
╭───────────────────── Traceback (most recent call last) 
──────────────────────╮
│ /home/u6dce01f3/Training/AI/GenAI/automatic/
launch.py │
│ :231 in <module> 
│
│ 
│
│ 230 
│
│ ❱ 231 uv, instance = start_server(immediate=True, server=None) 
│
│ 232 while True: 
│
│ 
│
│ /home/u6dce01f3/Training/AI/GenAI/automatic/
launch.py │
│ :171 in start_server 
│
│ 
│
│ 170 else: 
│
│ ❱ 171 uvicorn = server.webui(restart=not immediate) 
│
│ 172 if args.profile: 
│
│ 
│
│ /home/u6dce01f3/Training/AI/GenAI/automatic/
webui.py: │
│ 311 in webui
│ 310 start_common() 
│
│ ❱ 311 start_ui() 
│
│ 312 modules.sd_models.write_metadata() 
│
│ 
│
│ /home/u6dce01f3/Training/AI/GenAI/automatic/
webui.py: │
│ 229 in start_ui 
│
│ 
│
│ 228 timer.startup.record("before-ui") 
│
│ ❱ 229 shared.demo = modules.ui.create_ui(timer.startup) 
│
│ 230 timer.startup.record("ui") 
│
│ 
│
│ /home/u6dce01f3/Training/AI/GenAI/automatic/
modules/u │
│ i.py:908 in create_ui 
│
│ 
│
│ 907 with gr.Blocks(analytics_enabled=False) as 
control_interface: │
│ ❱ 908 from modules import ui_control 
│
│ 909 ui_control.create_ui() 
│
│ 
│
│ /home/u6dce01f3/Training/AI/GenAI/automatic/
modules/u │
│ i_control.py:3 in <module> 
│
│ 
│
│ 2 import gradio as gr 
│
│ ❱ 3 from modules.control import unit 
│
│ 4 from modules.control import processors # patrickvonplaten  controlnet_a │
│ 
│
│ /home/u6dce01f3/Training/AI/GenAI/automatic/
modules/c │
│ ontrol/unit.py:9 in <module> 
│
│ 
│
│ 8 from modules.control.units import t2iadapter 
│
│ ❱ 9 from modules.control.units import reference # pylint: 
disable=unused-i │
│ 10 
│
│ 
│
│ /home/u6dce01f3/Training/AI/GenAI/automatic/
modules/c │
│ ontrol/units/reference.py:15 in <module> 
│
│ 
│
│ 14 
│
│ ❱ 15 class ReferencePipeline(): 
│
│ 16 def __init__(self, pipeline: StableDiffusionXLPipeline | 
StableDiff │
│ 
│
│ /home/u6dce01f3/Training/AI/GenAI/automatic/
modules/c │
│ ontrol/units/reference.py:16 in ReferencePipeline 
│
│ 
│
│ 15 class ReferencePipeline(): 
│
│ ❱ 16 def __init__(self, pipeline: StableDiffusionXLPipeline | 
StableDiff │
│ 17 t0 = time.time() 
│
╰─────────────────────────────────────────────────────────────────────
─────────╯
TypeError: unsupported operand type(s) for |: 'type' and 'type'

@vladmandic

**Comment by vladmandic** [comment]
1. please do not tag me like that
2. always post version information
3. this is an issue, not a discussion item

having said all that, this is due to incompatibility with old python 3.9 and it has been fixed in dev branch.

**Comment by Disty0** [comment]
I have 3 issues with this.
1) You are using CUDA PyTorch with an Intel GPU.
2) You are using CUDA PyTorch with OpenVINO Backend.
3) You are using OpenVINO Backend with Intel Dev Cloud which is specifically for IPEX.

Wipe your current install and install the dev branch with --use-ipex.
